### PR TITLE
Correctly generate the feedback URL for remote share

### LINKED ISSUE #15589: File marked as shared but remote user cancelled the share
### Steps to reproduce
1. Upload a file (server A) and share it with a remote user in other server (Click in the share, and use something like "user@http://server/owncloud" to send the link - "user" must exists in the remote ownCloud server)
2. In the other server (server B), login as the user. A popup should appear, telling you that a file has been shared with you.
3. Cancel the sharing

### Expected behaviour
File shared inside server A isn't marked as shared (the remote user has cancelled it, so the file isn't shared)

### Actual behaviour
File is marked as shared

### Server configuration
**Operating system**: ubuntu 14.04

**Web server:** apache 2.4.7

**Database:** mysql

**PHP version:** 5.5.4

**ownCloud version:** OC 8.0.3RC1

**Updated from an older ownCloud or fresh install:** fresh install

### PR DESCRIPTION
**Pull request by nickvergessen**
The trailing slash was added in c78e3c4a7fa1d2f474ab58551e67a50e093f6ed8
to correctly generate the encryption keys

Fix #15589 

@jvillafanez @schiesbn

### CODE REVIEW

**Comment by nickvergessen** [comment]
will push again, after jenkins reported back

**Comment by karlitschek** [comment]
:+1:

**Comment by nickvergessen** [comment]
@schiesbn can you review this :)

**Comment by DeepDiver1975** [comment]
@nickvergessen can we unit test this somehow please? The next change will break this again unrecognized. THX

**Comment by nickvergessen** [comment]
Added tests and also fixed one more problem :speak_no_evil:

File: tests/lib/share/share.php [hunk 1]
```diff
@@ -105,6 +105,12 @@ protected function tearDown() {
 		parent::tearDown();
 	}
 
+	protected function setHttpHelper($httpHelper) {
+		\OC::$server->registerService('HTTPHelper', function ($c) use ($httpHelper) {
```
Comment (nickvergessen) [inline tests/lib/share/share.php]: The parameter $c is not used and could be removed.

File: lib/private/share/share.php [hunk 2]
```diff
@@ -2392,15 +2392,17 @@ private static function removeProtocolFromUrl($url) {
 	 *
 	 * @param string $url
 	 * @param array $fields post parameters
-	 * @return bool
+	 * @return array
```
Comment (nickvergessen) [inline lib/private/share/share.php]: the return was always an array, im just fixing the doc

**Comment by MorrisJobke** [comment]
For me the share is still present in the web UI of the sharee. :(

My steps:
* share a file with a remote user via USERNAME@remote.address
* login as USERNAME at remote.address
* cancel share

**Comment by nickvergessen** [comment]
Ah, it works on the normal "unshare" now, after you accepted it once. Will try to find the root cause of this one aswell

**Comment by nickvergessen** [comment]
Updated

works as expected now for insta decline and accept-decline scenarios

please review @PVince81 @MorrisJobke @rullzer

**Comment by PVince81** [comment]
Tested, works :+1: 

Unsharing from self or refusing the share properly removes it from the sharers' shares list

**Comment by MorrisJobke** [comment]
Works now :+1:

**Comment by MorrisJobke** [comment]
@karlitschek @nickvergessen Backport?

**Comment by karlitschek** [comment]
please backport

**Comment by DeepDiver1975** [comment]
@nickvergessen please submit a pr against stable8 - THX

**Comment by nickvergessen** [comment]
Done, but half of the test coverage is missing, because the PRs which made up the coverage where not backported

**Pull request merged**